src/components/Chart/ChartContext.js

```diff
@@ -0,0 +1,215 @@
+
+  const data = values
+    .filter(getDataFilter(mergedProps.filter))
+    .filter(hasValues)
+    .map(normalizeData(mergedProps.x, xNormalizer))
+    .map(categorizeData(mergedProps.category))
+
+  const colorAccessor = mergedProps.color
+    ? d => d.datum[mergedProps.color]
+    : d => d.category
+
+  const colorValues = []
+    .concat(data.map(colorAccessor))
+    .concat(mergedProps.colorLegendValues)
+    .filter(Boolean)
+    .filter(deduplicate)
+
+  const xValuesUnformatted = data
+    .map(xAccessor)
+    .concat(props.xTicks ? props.xTicks.map(xNormalizer) : [])
+    .concat(getAnnotationsXValues(xAnnotations, xNormalizer))
+    .filter(deduplicate)
+
+  const processedData = dataProcesser[type]({
+    props: mergedProps,
+    data,
+    colorAccessor,
+    colorValues,
+    xValuesUnformatted,
+    xFormat,
+    xParser,
+    xParserFormat,
+    xSort,
+    xNormalizer
+  })
+
+  const chartContextObject = {
+    ...processedData,
+    colorAccessor,
+    colorValues,
+    xNormalizer
+  }
+
+  return (
+    <ChartContext.Provider value={chartContextObject}>
+      {mergedProps.children}
+    </ChartContext.Provider>
+  )
+}
+
+export const defaultProps = {
+  TimeBar: {
+    x: 'year',
+    xScale: 'time',
+    xBandPadding: 0.25,
+    timeParse: '%Y',
+    numberFormat: 's',
+    height: 240,
+    padding: 50,
+    unit: '',
+    xUnit: '',
+    colorLegend: true,
+    xIntervalStep: 1,
+    yAnnotations: [],
+    xAnnotations: [],
+    columns: 1,
+    minInnerWidth: 240
+  },
+  Line: {
+    x: 'year',
+    xScale: 'time',
+    yScale: 'linear',
+    timeParse: '%Y',
+    timeFormat: '%Y',
+    numberFormat: '.0%',
+    zero: true,
+    unit: '',
+    startValue: false,
+    endValue: true,
+    endLabel: true,
+    endDy: '0.3em',
+    minInnerWidth: 110,
+    columns: 1,
+    height: 240,
+    yNice: 3
+  },
+  Slope: {
+    x: 'year',
+    xScale: 'ordinal',
+    yScale: 'linear',
+    timeParse: '%Y',
+    timeFormat: '%Y',
+    numberFormat: '.0%',
+    zero: true,
+    unit: '',
+    startValue: true,
+    endValue: true,
+    endLabel: false,
+    endDy: '0.3em',
+    minInnerWidth: 90,
+    columns: 2,
+    height: 240,
+    yNice: 3
+  }
+}
+
+const axisPropType = PropTypes.shape({
+  ticks: PropTypes.array.isRequired,
+  format: PropTypes.func.isRequired,
+  axisFormat: PropTypes.func.isRequired
+})
+
+const propTypes = {
+  groupedData: PropTypes.array.isRequired,
+  groupPosition: PropTypes.shape({
+    y: PropTypes.func.isRequired,
+    x: PropTypes.func.isRequired,
+    titleHeight: PropTypes.number // timebar only
+  }).isRequired,
+  height: PropTypes.number.isRequired,
+  innerWidth: PropTypes.number.isRequired,
+  x: PropTypes.func.isRequired,
+  xAxis: axisPropType.isRequired,
+  xDomain: PropTypes.array.isRequired,
+  y: PropTypes.func.isRequired,
+  // inconsistent
+  colorValues: PropTypes.array,
+  colorLegend: PropTypes.bool,
+  colorLegendValues: PropTypes.array,
```

Comment:
I re-ordered a bit here, I feel like color stuff should be cleaned up and exclusively use via context.

E.g. the context could just provide a `color(datum) => cssColorValue` function (accessor and mapper in one) and a `colorLegendValues` (array of `{color: string, label: string}`) and empty if disabled via props (`colorLegend = false`) or unnecessary (e.g. only one color, or inline line labels).

```suggestion
  color: PropTypes.func.isRequired,
  colorLegendValues: PropTypes. arrayOf(PropTypes.shape({
    color: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired
  })).isRequired,
```
```